test/unit_tests/test_interface.py

```diff
@@ -48,7 +50,15 @@ def test_fadump_enabled(self):
         self._check_properties_changed("FadumpEnabled", True)
         self.assertEqual(self._interface.FadumpEnabled, True)
 
-    def test_reserved_memory(self):
+    @patch("com_redhat_kdump.service.kdump.KdumpService.check_reserved_memory", return_value="256")
+    def test_reserved_memory(self, _mock_read):
         self._interface.ReservedMemory = "256"
         self._check_properties_changed("ReservedMemory", "256")
         self.assertEqual(self._interface.ReservedMemory, "256")
+
+    @patch("com_redhat_kdump.common.getMemoryBounds", return_value=(500, 800, 1))
+    def test_check_reserved_memory(self, _mock_read):
+        self._service._lower , self._service._upper, self._service._step = common.getMemoryBounds()
+        self.assertEqual(self._service.check_reserved_memory("900"), "800")
+        self.assertEqual(self._service.check_reserved_memory("400"), "500")
+        self.assertEqual(self._service.check_reserved_memory("600"), "600")
```

Comment:
For this test, you should patch `com_redhat_kdump.service.kdump.getMemoryBounds` instead because you should patch a function where it's used (imported) instead of where it's defined. Though unittest doesn't support parameterized tests,  you can still simplify the code a bit,
```python
    @patch("com_redhat_kdump.service.kdump.getMemoryBounds", return_value = (500, 800, 1))
    def test_check_reserved_memory(self, mocker):
        service1 = KdumpService()
        for memory, reserved in [("900", "800"), ("400", "500"), ("600", "600")]:
            self.assertEqual(service1.check_reserved_memory(memory), reserved)
```